test/distributed/test_c10d.py

```diff
@@ -4639,6 +4640,48 @@ def test_broadcast_coalesced_gloo_cpu(self):
         for root_rank in ranks:
             self._test_broadcast_coalesced(process_group, device, root_rank)
 
+    @requires_gloo()
+    def test_pass_gloo_options(self):
+        pg_opts = c10d.ProcessGroupGloo.Options()
+        pg_opts.devices = [create_device(interface=LOOPBACK)]
+        pg_opts.timeout = timedelta(seconds=1)
+        pg_opts.threads = 2
```

Comment:
Maybe we want to test what happens if devices is set but threads are not, and if threads are set but devices or not?  Although doesn't seem like too big of an issue - we are only exposing timeout so this logic should be okay.